### PR TITLE
Store indicator component mapping as part of Visualizer's store subscription

### PR DESCRIPTION
**Pull request by Wumpf**
### What
 
* Kinda part of #4388
* Continuing the spirit of #4604
    * Exact same reasoning and motiviation applies! 
    * This time this actually has clear positive perf implications: Evaluation of `identify_entities_per_system_per_class` goes from 1ms to 0.5ms on `opf --no-frames`
        * but again in the "heuristic evaluation is part of query"-world this is a _huge_ speedup

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4608/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4608/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4608/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4608)
- [Docs preview](https://rerun.io/preview/43ecf8119748c2370d859cade05837fd62f9ffe9/docs) 
- [Examples preview](https://rerun.io/preview/43ecf8119748c2370d859cade05837fd62f9ffe9/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_viewport/src/space_view_heuristics.rs [hunk 1]
```diff
@@ -435,34 +434,18 @@ pub fn is_entity_processed_by_class(
...
-        .into_iter()
-        .collect();
-    for part in parts.iter() {
-        if part.heuristic_filter(store, ent_path, ctx, query, &components) {
+    let empty_entity_set = IntSet::default();
+    for (id, visualizer) in parts.iter_with_identifiers() {
+        let entites_with_matching_indicator = ctx
+            .entities_with_matching_indicator_per_visualizer
+            .get(&id)
+            .unwrap_or(&empty_entity_set);
```
Comment (jleibs) [inline crates/re_viewport/src/space_view_heuristics.rs:446]: If this is an empty set, do we still need to run: `visualizer.heuristic_filter`, or can we just return false directly?

File: crates/re_viewer_context/src/viewer_context.rs [hunk 2]
```diff
@@ -35,6 +37,10 @@ pub struct ViewerContext<'a> {
     /// Mapping from class and system to entities for the store
     pub entities_per_system_per_class: &'a EntitiesPerSystemPerClass,
```
Comment (jleibs) [inline crates/re_viewer_context/src/viewer_context.rs:38]: Nit: this needs a more specific name.

**Review by jleibs — APPROVED** [review]
Makes sense. Definitely seems worth computing with the subscriber.

**Review by jleibs — APPROVED** [review]
Makes sense. Definitely seems worth computing with the subscriber.

File: crates/re_viewer_context/src/viewer_context.rs (hunk 2 above)
Comment (Wumpf) [reply crates/re_viewer_context/src/viewer_context.rs:38]: which is why it is renamed in the next pr 😄

File: crates/re_viewport/src/space_view_heuristics.rs (hunk 1 above)
Comment (Wumpf) [reply crates/re_viewport/src/space_view_heuristics.rs:446]: Hmm practically yes. But technically `heuristic_filter` decides whehter to use `entites_with_matching_indicator` in the first place 🤔
I'm a bit undecided. leaving it for now

**Pull request merged**